tests/ui/object-safety/call-when-assoc-ty-is-sized.rs

```diff
@@ -0,0 +1,25 @@
+// check-pass
+// revisions: current next
+//[next] compile-flags: -Ztrait-solver=next
+
+trait Foo {
+    type Bar<'a>
+    where
+        Self: Sized;
+
+    fn test(&self);
+}
+
+impl Foo for () {
+    type Bar<'a> = () where Self: Sized;
+
+    fn test(&self) {}
+}
+
+fn test(x: &dyn Foo) {
+    x.test();
```

Comment:
This is a change in stable behaviour (compiles now, errors on nightly) and needs an FCP

Reply:
I guess that's what you mean by "doesn't need re-litigation". Ugh... I agree, but we have procedures out of a reason, too.

Reply:
Start an FCP then 🤷